twoliter/src/lock.rs

```diff
@@ -28,3 +38,234 @@ pub(crate) struct LockedImage {
+                            "cannot use two different sdks from two different vendors, if you meant to override the vendor you must specify [sdk] in Twoliter.toml ({} != {})",
+                            sdk.vendor,
+                            metadata.sdk.vendor,
+                        );
+                    }
+                } else {
+                    // The sdk could be from a different vendor so find it here
+                    let sdk_vendor = vendor_table.get(&metadata.sdk.vendor).context(format!(
+                        "vendor '{}' was not specified in Twoliter.toml, and is required for build",
+                        metadata.sdk.vendor
+                    ))?;
+                    sdk = Some(LockedImage {
+                        name: metadata.sdk.name.clone(),
+                        version: metadata.sdk.version.clone(),
+                        vendor: metadata.sdk.vendor.clone(),
+                        source: format!(
+                            "{}/{}:v{}",
+                            sdk_vendor.registry, metadata.sdk.name, metadata.sdk.version
+                        ),
+                    });
+                }
+
+                // Build the LockedImage for this
+                let locked = LockedImage {
+                    name: metadata.name.clone(),
+                    version: metadata.version.clone(),
+                    vendor: image.vendor.clone(),
+                    source: format!(
+                        "{}/{}:v{}",
+                        vendor.registry, metadata.name, metadata.version
+                    ),
+                };
+                // Now for the dependencies in this kit add them back if they are not already in the selected
+                // list
+                for kit in metadata.kit.iter() {
+                    next.insert(LockedImage::hash_image(kit));
+                }
+                kit.insert(locked);
+            }
+            tangential_images = next.clone();
+        }
+
+        Ok(Lock {
+            schema_version: project.schema_version(),
+            release_version: project.release_version().to_string(),
+            kit: kit.iter().cloned().collect(),
+            sdk: sdk.context("could not resolve a sdk version from dependencies")?,
+            digest: project.digest()?,
+        })
+    }
+
+    async fn find_kit(vendor: &Vendor, image: &LockedImage) -> Result<ImageMetadata> {
+        // First construct the initial image-uri to check, do not rely on image.source as it is currently in a shape
+        // for hash comparisons
+        let image_uri = format!("{}/{}:v{}", vendor.registry, image.name, image.version);
+
+        // Now inspect the manifest list
+        let manifest_output = docker_command(["manifest", "inspect", image_uri.as_str()]).await?;
+        // If this hit an error the kit does not exist
+        ensure!(
+            manifest_output.status.success(),
+            "no such kit foud at uri {}",
```

Comment:
nit:
```suggestion
            "no such kit found at uri {}",
```

but also you should indicate the kit's name / vendor / version so the error message is more clear about why we checked that particular URI